### PR TITLE
feat: Make RegistryServer writable

### LINKED ISSUE #4216: Make RegistryServer writable
**Is your feature request related to a problem? Please describe.**
[RegistryServer](https://github.com/feast-dev/feast/blob/master/sdk/python/feast/registry_server.py) is currently read-only. It needs to be writable as well to fit into the security model #4198.

### PR DESCRIPTION
**Pull request by tokoko**
# What this PR does / why we need it:
- This PR adds apply/delete endpoints to the grpc registry server, also makes necessary changes in `remote` registry that call those endpoints.

# Which issue(s) this PR fixes:
Fixes #4216

### CODE REVIEW

File: sdk/python/tests/integration/registration/test_feature_store.py [hunk 1]
```diff
@@ -11,68 +11,21 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import os
```
Comment (franciscojavierarceo) [inline sdk/python/tests/integration/registration/test_feature_store.py:14]: There's a lot being deleted here. Can you update the PR notes to explain why? I'm not fully familiar with the usage of all of these tests so I'd like to make sure I understand the implications of removing all of these tests (along with the other files).

File: sdk/python/tests/integration/registration/test_universal_registry.py [hunk 2]
```diff
@@ -187,19 +191,70 @@ def sqlite_registry():
...
+
+# sql_fixtures = [
+#     "pg_registry",
+#     "mysql_registry",
+#     "sqlite_registry",
+# ]
+
+
 @pytest.mark.integration
-@pytest.mark.parametrize(
```
Comment (franciscojavierarceo) [inline sdk/python/tests/integration/registration/test_universal_registry.py:191]: Same here

File: sdk/python/tests/unit/infra/registry/test_remote.py [hunk 3]
```diff
@@ -1,69 +0,0 @@
-import assertpy
```
Comment (franciscojavierarceo) [inline sdk/python/tests/unit/infra/registry/test_remote.py:1]: Same here

File: sdk/python/tests/unit/test_registry_server.py [hunk 4]
```diff
@@ -1,60 +0,0 @@
-import assertpy
```
Comment (franciscojavierarceo) [inline sdk/python/tests/unit/test_registry_server.py:1]: Same here

**Review by franciscojavierarceo — CHANGES_REQUESTED** [review]
Can you add more details to the PR notes for this? Looks like we're removing a lot of tests and I'd like to make sure I understand the consequences of doing so as I'm not familiar with the usage of those tests.

**Comment by tokoko** [comment]
Updated the PR description, let me know if it needs more clarity.

**Comment by franciscojavierarceo** [comment]
Thanks for updating the PR description. 

Can you separate the removal of the tests into another PR? It'd be much cleaner in case we need to revert it.

**Comment by tokoko** [comment]
@franciscojavierarceo cut down on the scope of the PR, reverted all test removals.

**Review by franciscojavierarceo — APPROVED** [review]
thank you!

**Pull request merged**